### DISCUSSION TITLE
Is there any difference in behavior when reusing the same `Arbitrary` instance vs. using distinct instances?

### DISCUSSION THREAD

**Discussion by jmartinezmaes** [dubzzz/fast-check#3530] [Q&A] [answered]
## 💬 Question and Help

Does this:

```ts
import * as fc from "fast-check";

const strUpper = fc.string().map((s) => s.toUpperCase());

fc.assert(fc.property(strUpper, strUpper, (x, y) => { /* asserts here */ }));
```

and this:

```ts
import * as fc from "fast-check";

function strUpper(): fc.Arbitrary<string> {
    return fc.string().map((s) => s.toUpperCase());
}

fc.assert(fc.property(strUpper(), strUpper(), (x, y) => { /* asserts here */ }));
```

have any difference in functionality?

**Comment by dubzzz** [comment, marked as answer]
I'd say there is no difference at all. All arbitraries are stateless and re-using or not the instance has no impact on fast-check's behaviour, it's supposed to behave fully the same.

The function approach is closer to the APIs of fast-check, but while it makes sense in the library to offer configurable arbitraries it's mostly up to the user when it's for user specific arbitraries.
